cratetorrent/src/metainfo.rs

```diff
@@ -14,6 +14,20 @@ pub use serde_bencode::Error as BencodeError;
 
 pub(crate) type Result<T> = crate::error::Result<T, MetainfoError>;
 
+/// The protocol that the tracker is using
+#[derive(Clone)]
+pub enum NetProtocol {
+    UDP,
+    HTTP,
+}
+
+
+#[derive(Clone)]
+pub struct TrackerUrl {
+    pub url: Url,
+    pub protocol: NetProtocol,
```

Comment:
I don't think this is necessary. The `Url` type already contains the scheme.

Reply:
The Url struct is reqwest only, so it only works for HTTP, not for UDP